Sming/Wiring/WHashMap.h

```diff
@@ -91,16 +116,18 @@ class HashMap
     || @description
     || | An indexer for accessing and assigning a value to a key
     || | If a key is used that exists, it returns the value for that key
-    || | If there exists no value for that key, the key is added
+    || | If there exists no value for that key, a nil value is returned
     || #
     ||
     || @parameter key the key to get the value for
     ||
     || @return The const value for key
     */
-    const V& operator[](const K key) const
+    const V& operator[](const K& key) const
```

Comment:
Another test comes to my mind:
```
       HttpHeaders headers;

	debugf("Non-Existent Header: [%s]", headers["Non-Existent"]);

	headers["X-Auth-Header"] = "Value Data";
	headers["x-Auth-header"] = "Value Data New";

	debugf("Header: %s", headers["X-AUTH-HEADER"].c_str());

	headers.remove("x-auth-header");

	debugf("Headers count: %d", headers.count());
```

If I am not mistaken the expected output with those changes should be:
```
Header: Value Data New

Headers count: 0
```
right? 
If the answer is no - then add a comment describing this
if the answer is yes - check if you are getting the expected result and if not make sure to fix it :)

Reply:
Answer is no, because headers is not a const object.

```
	HttpHeaders headers;

	const HttpHeaders& constHeaders = headers;
	debugf("Non-Existent Header (const): \"%s\"", constHeaders["Non-Existent"].c_str());
	debugf("  count: %d", headers.count());

	debugf("Non-Existent Header (non-const): \"%s\"", headers["Non-Existent"].c_str());
	debugf("  count: %d", headers.count());

	headers["X-Auth-Header"] = "Value Data";
	debugf("set(\"X-Auth-Header\")");
	debugf("  count: %d", headers.count());

	headers["x-Auth-header"] = "Value Data New";
	debugf("set(\"x-Auth-Header\")");
	debugf("  count: %d", headers.count());

	debugf("Header[\"X-AUTH-HEADER\"]: \"%s\"", headers["X-AUTH-HEADER"].c_str());
	debugf("  count: %d", headers.count());

	headers.remove("x-auth-header");
	debugf("headers.remove(\"x-auth-header\")");
	debugf("  count: %d", headers.count());
```

```
1916918089 Non-Existent Header (const): ""
1916918089   count: 0
1916918089 Non-Existent Header (non-const): ""
1916918089   count: 1
1916918089 set("X-Auth-Header")
1916918089   count: 2
1916918089 set("x-Auth-Header")
1916918089   count: 2
1916918089 Header["X-AUTH-HEADER"]: "Value Data New"
1916918089   count: 2
1916918089 headers.remove("x-auth-header")
1916918089   count: 1
```

Reply:
Perfect, thanks!